pkg/foxtrot/auth.go

```diff
@@ -0,0 +1,39 @@
+package foxtrot
+
+import (
+	"context"
+	"errors"
+
+	"foxygo.at/s/errs"
+	"golang.org/x/crypto/bcrypt"
+)
+
+var (
+	errAuth         = errors.New("user authentication error")
+	errPasswordHash = errors.New("password hash creation err")
+)
+
+type authenticator struct {
+	db *db
+}
+
+func (a *authenticator) register(ctx context.Context, u *User, password string) error {
```

Comment:
my feeling is that the user parameter should be a string. registration inputs will be username and password. a User struct would not exist at this point, yes?

Reply:
not sure. would you in a signup flow do all: username, nickname, fullname, avatar, password, phone number, address, relationship status?

Reply:
if yes, then i think it is ok to expect a user struct as input.

Reply:
relationship status? what are we building here? ;-)

I'm not fussed, but my uninformed expectation is that those things would not be provided on registration.